### PR TITLE
Add Convex to the demo

### PR DESCRIPTION
**Pull request by preslavle**
Add Convex to the Edge <> Data latency demo using the ConvexHttpClient.

I have checked in `convex.json` and `.env` that hook the demo to an existing deployment. To create a new deployment:
1. `rm -rf convex.json`
2. `npm install`
3. `npx convex init`
4. `npx convex deploy`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,19 +6,19 @@ https://edge-data-latency.vercel.app
...
 ## Providers
 
+- Convex (SDK)
+- Fauna
 - Grafbase (GraphQL)
 - PlanetScale (Kysely + Serverless SDK)
 - Shopify (Storefront GraphQL API)
 - Supabase (supabase-js)
 - Xata (TypeScript SDK)
 - Upstash (SDK)
```
Comment (sujayakar) [inline README.md]: upstash before xata alphabetically?

File: convex/schema.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { defineSchema, defineTable, s } from "convex/schema";
+
+export default defineSchema({
+    employees: defineTable({
+      emp_no: s.number(),
+      first_name: s.string(),
+      last_name: s.string(),
```
Comment (sujayakar) [inline convex/schema.ts]: would camelCase be more idiomatic? do they use snake case in their schema elsewhere?

File: pages/api/convex-global.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      headers: {
+        "x-edge-is-cold": start === time ? "1" : "0",
+      },
+    }
+  );
+}
+
+// convert a query parameter to a number
+// also apply a min and a max
```
Comment (sujayakar) [inline pages/api/convex-global.ts:45]:
```suggestion
// Convert a query parameter to a number, clamping to a min and max.
```

File: pages/api/convex-global.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  for (let i = 0; i < count; i++) {
+    const limit = 10;
+    data = await convex.query("employees")(limit);
+  }
+
+  return Response.json(
+    {
+      data,
+      queryDuration: Date.now() - time,
+      invocationIsCold: start === time,
```
Comment (sujayakar) [inline pages/api/convex-global.ts:32]: is this too aggressive/do they do the same detection elsewhere? would this return false if module loading/etc. takes more than 1ms?

File: pages/api/xata-global.ts [hunk 5]
```diff
@@ -46,6 +46,9 @@ export default async function api(req: Request) {
   for (let i = 0; i < count; i++) {
     data = await xata.db.employees
       .select(["emp_no", "first_name", "last_name"])
+      .filter({
+        last_name: {$startsWith: String.fromCharCode(65 + i)}
+      })
```
Comment (sujayakar) [inline pages/api/xata-global.ts]: should we revert this for now? we can send it as a separate PR.

**Review by sujayakar — COMMENTED** [review]
awesome stuff!

File: pages/api/xata-global.ts (hunk 5 above)
Comment (preslavle) [reply pages/api/xata-global.ts]: Oh, yeah, will revert this.

File: convex/schema.ts (hunk 2 above)
Comment (preslavle) [reply convex/schema.ts]: They do in SQL. But I can do camelCase here.

File: pages/api/convex-global.ts (hunk 4 above)
Comment (preslavle) [reply pages/api/convex-global.ts:32]: Yeah. This is what they do everywhere else :(. It can even flake if it takes less than 1ms but lets you move to the next second... I will keep it as is to be consistent.

File: pages/api/convex-global.ts (hunk 3 above)
Comment (preslavle) [reply pages/api/convex-global.ts:45]: I copy & pasted this but sure.

**Review by sujayakar — APPROVED** [review]
